storage/src/rocksdb_client.cpp

```diff
@@ -646,6 +647,121 @@ void Client::removeAllCheckpoints() const {
   }
 }
 
+Client::RocksDbStorageMetrics::RocksDbStorageMetrics(const std::vector<::rocksdb::Tickers> &tickers,
+                                                     const Client &owningClient)
+    : StorageMetricsBase({"storage_rocksdb", std::make_shared<concordMetrics::Aggregator>()},
+                         update_metrics_interval_msec_),
+      owning_client_(owningClient),
+      total_db_disk_size_(metrics_.RegisterAtomicGauge("storage_rocksdb_total_db_disk_size", 0)),
+      all_mem_tables_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_mem_tables_ram_usage", 0)),
+      all_column_families_block_cache_ram_usage_(
+          metrics_.RegisterAtomicGauge("storage_rocksdb_column_families_block_cache_ram_usage", 0)),
+      indexes_and_filters_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_indexes_and_filter_ram_usage", 0)),
+      rocksdb_total_ram_usage_(metrics_.RegisterAtomicGauge("storage_rocksdb_total_ram_usage", 0)) {
+  for (const auto &pair : ::rocksdb::TickersNameMap) {
+    if (std::find(tickers.begin(), tickers.end(), pair.first) != tickers.end()) {
+      auto metric_suffix = pair.second;
+      std::replace(metric_suffix.begin(), metric_suffix.end(), '.', '_');
+      active_tickers_.emplace(pair.first, metrics_.RegisterAtomicGauge("storage_" + metric_suffix, 0));
+    }
+  }
+  metrics_.Register();
+}
+
+void Client::RocksDbStorageMetrics::setMetricsDataSources(std::shared_ptr<::rocksdb::SstFileManager> sourceSstFm,
+                                                          std::shared_ptr<::rocksdb::Statistics> sourceStatistics) {
+  sstFm = sourceSstFm;
+  statistics = sourceStatistics;
+
+  update_metrics_ =
+      std::make_unique<concord::util::PeriodicCall>([this]() { updateMetrics(); }, update_metrics_interval_msec_);
+}
+
+// periodically running function to update metrics. this func and its sub-funcs aren't
+// thread safe. It's the Client responsibility to temporarily disable this periodic func
+// if configs change during runtime.
+void Client::RocksDbStorageMetrics::updateMetrics() {
+  static size_t entry_count{0};
+
+  // if isn't initialized yet or disabled - return
+  if (!sstFm || !statistics) return;
+
+  // we don't update mem usage every call since it's pretty heavy on resources and isn't needed in a very high frequency
+  if (entry_count++ % update_mem_usage_metrics_factor_ == 0) {
```

Comment:
This func runs every update_metrics_interval_msec_ but we want it to actually update the metrics in a lower rate so we have a factor. it will update the metrics every 'update_metrics_interval_msec_ * update_mem_usage_metrics_factor_' . not sure I see the issue here...